### PR TITLE
chore: update Nvidia driver and device plugin

### PR DESCRIPTION
**Pull request by jongiddy**
**What type of PR is this?**

Chore

**What this PR does/ why we need it**:
Update device plugin and remove container-based Nvidia driver.

**Which issue(s) this PR fixes**:
https://jira.d2iq.com/browse/D2IQ-75523
https://jira.d2iq.com/browse/D2IQ-75524

**Special notes for your reviewer**:

Tested by running this job on a cluster using the updated chart:
```yaml
apiVersion: batch/v1
kind: Job
metadata:
  name: gpu-test
spec:
  template:
    spec:
      containers:
      - name: cuda-vector-add
        image: k8s.gcr.io/cuda-vector-add:v0.1
        resources:
          limits:
            nvidia.com/gpu: 1
      tolerations:
        - key:      "dedicated"
          operator: "Equal"
          value:    "gpu-worker"
          effect:   "NoExecute"
      restartPolicy: Never
  backoffLimit: 4
```
This matches the test at https://github.com/mesosphere/konvoy/blob/f9f049ac47f5f770602d3a99c44c74eb44d4aa5d/tests/e2e/functions.go#L1859-L1918

**Does this PR introduce a user-facing change?**:

```release-note
Updated Nvidia addon to support driver v460
```

**Checklist**

* [x] *If a chart is changed, the chart version is correctly incremented.*
* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

File: staging/nvidia/Chart.yaml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 apiVersion: v1
 name: nvidia
-description: Nvidia GPU driver and device plugin for running Nvidia GPU
+description: Nvidia GPU device plugin for running Nvidia GPU
 keywords:
   - gpu
   - nvidia
-version: 0.3.6
+version: 0.3.7
```
Comment (mhrabovcin) [inline staging/nvidia/Chart.yaml:7]: Shouldn't we bump this version to `0.4.x` as we're removing a significant part of this addon - the driver?

Comment (jongiddy) [reply staging/nvidia/Chart.yaml:7]: Sounds reasonable. Before I do, are there any actual behavior differences in changing more-major parts of the version?

Comment (tillt) [reply staging/nvidia/Chart.yaml]: I wouldn't know any such differences @jongiddy

**Review by tillt — APPROVED** [review]
0.9.0 is indeed more recent than 1.0.0-beta4 - mind boggling :D

**Review by mhrabovcin — APPROVED** [review]

**Pull request merged**